src/metric.f90

```diff
@@ -94,7 +94,7 @@ subroutine metric
  case(2)
 
   allocate( sinc, sini, cosc, cosi, mold=x2 )
-  do j = js-2, je+2
+  do j = gjs-2, gje+2 ! TODO: might need to set other arrays over the 'g' indices too...
```

Comment:
POSSIBLE BUG: Other loops may also need to be changed?

Reply:
"ensure sinc(j) is set across gjs:gje otherwise might be zero when used
in computing hg123 in gravity.f90 line 959:963"
https://github.com/ryosuke-hirai/HORMONE/pull/34/commits/3ee8b533c7e95990fedfd5bde64a54e2ac8ba472

Reply:
`gjs:gje` should be equal to `js:je` in spherical coordinates (same for `gks:gke`). The only direction the grid can be extended in spherical coordinates is radially (`gie`).
Same applies for `gjs:gje` in cylindrical coordinates.

These conditions are ensured in `checksetup.f90` line 150

Reply:
Also, I am doing tests in the 3D spherical case to not use `hg` variables any more but reuse the `cg%` variables instead.
I plan to eventually get rid of all the `hg` variables or just make them pointers to `cg%` instead.

Reply:
Ahh, ok I see why I was running into problems. It's because we haven't done the domain decomposition for the 'gravity' grid yet. So each MPI task is holding the full array, if it was allocated using the 'g' indices.

@conradtchan I think we should implement the domain decomp as part of this PR. i.e. ensure that is,ie,... etc, are consistent with gis,gie,...etc

Reply:
Reverted in https://github.com/ryosuke-hirai/HORMONE/pull/34/commits/f2b531bffde264a9ef54e4a89e306639793bffbe